File: jan2/jan2.cs
using System;
using System.Linq;
using System.Text.RegularExpressions;

class Program {
    public static bool IsPalindrome(string str) {
        string formattedStr = Regex.Replace(str.ToLower(), @"[^a-zA-Z0-9]", "");
        string reversedStr = new string(formattedStr.Reverse().ToArray());
        return formattedStr == reversedStr;
    }

    static void Main() {
        string input = "A man, a plan, a canal, Panama";
        bool result = IsPalindrome(input);
        Console.WriteLine(result);
    }
}

File: jan2/jan2.js
function isPalindrome(str) {
    const formattedStr = str.toLowerCase().replace(/[^a-zA-Z0-9]/g, '');
    const reversedStr = formattedStr.split('').reverse().join('');
    return formattedStr === reversedStr;
}

const input = "A man, a plan, a canal, Panama";
const result = isPalindrome(input);
console.log(result);
